src/course/course.service.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Inject,
+  Injectable,
+  Scope,
+  Request,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Course } from './entities/course.entity';
+import { Repository } from 'typeorm';
+import { CreateCourseDto } from './dtos/create-course.dto';
+import { User } from 'src/user/entities/user.entity';
+import { InstructorService } from 'src/instructor/instructor.service';
+import { UpdateCourseDto } from './dtos/update-course.dto';
+
+@Injectable()
+export class CourseService {
+  constructor(
+    @InjectRepository(Course)
+    private readonly courseRepository: Repository<Course>,
+    private readonly instructorService: InstructorService,
+  ) {}
+
+  findAll(): Promise<Course[]> {
+    return this.courseRepository.find();
+  }
+
+  findOne(id: number): Promise<Course> {
+    return this.courseRepository.findOneBy({ id });
+  }
+
+  findAllByUserId(userId: number): Promise<Course[]> {
+    return this.courseRepository.findBy({
+      instructor: { user: { id: userId } },
```

Comment:
actually it will join before search, like it join course and user first then it find by user id. Not so sure remember but you can help to test to verify it, if can, simply find by where {user_id: userId)